api/v1alpha1/ext_auth_types.go

```diff
@@ -0,0 +1,111 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import (
+	gwapiv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+// ExtAuthServiceType specifies the types of External Authorization.
+// +kubebuilder:validation:Enum=GRPC;HTTP
+type ExtAuthServiceType string
+
+const (
+	// GRPC external authorization service.
+	GRPC ExtAuthServiceType = "GRPC"
+
+	// HTTP external authorization service.
+	HTTP ExtAuthServiceType = "HTTP"
+)
+
+// ExtAuth defines the configuration for External Authorization.
+type ExtAuth struct {
+	// Type decides the type of External Authorization.
+	// Valid ExtAuthServiceType values are "GRPC" or "HTTP".
+	//
+	// +unionDiscriminator
+	Type ExtAuthServiceType `json:"type"`
+
+	// GRPCService defines the gRPC External Authorization service
+	// Only one of GRPCService or HTTPService may be specified.
+	GRPCService *GRPCExtAuthService `json:"grpcService,omitempty" yaml:"grpcService"`
```

Comment:
simply with `grpc`?

Reply:
No strong opinion, but GRPCService seems clearer to me.

Reply:
+1 to `grpc` and `http`